feature_engine/discretisation/increasing_width.py

```diff
@@ -0,0 +1,134 @@
+from typing import List, Optional, Union
+
+import pandas as pd
+import numpy as np
+
+from feature_engine._docstrings.fit_attributes import (
+    _feature_names_in_docstring,
+    _n_features_in_docstring,
+    _variables_attribute_docstring,
+)
+from feature_engine._docstrings.init_parameters import _variables_numerical_docstring
+from feature_engine._docstrings.methods import _fit_transform_docstring
+from feature_engine._docstrings.substitute import Substitution
+from feature_engine._variable_handling.init_parameter_checks import (
+    _check_init_parameter_variables,
+)
+from feature_engine.discretisation.base_discretiser import BaseDiscretiser
+
+
+@Substitution(
+    return_object=BaseDiscretiser._return_object_docstring,
+    return_boundaries=BaseDiscretiser._return_boundaries_docstring,
+    binner_dict_=BaseDiscretiser._binner_dict_docstring,
+    fit=BaseDiscretiser._fit_docstring,
+    transform=BaseDiscretiser._transform_docstring,
+    variables=_variables_numerical_docstring,
+    variables_=_variables_attribute_docstring,
+    feature_names_in_=_feature_names_in_docstring,
+    n_features_in_=_n_features_in_docstring,
+    fit_transform=_fit_transform_docstring,
+)
+class IncreasingWidthDiscretiser(BaseDiscretiser):
+    """
+    The IncreasingWidthDiscretiser() divides continuous numerical variables into
+    intervals of increasing width with equal increments. Note that the
+    proportion of observations per interval may vary.
+
+    The size of the interval will follow geometric progression.
+
+    The IncreasingWidthDiscretiser() works only with numerical variables.
+    A list of variables can be passed as argument. Alternatively, the discretiser
+    will automatically select all numerical variables.
+
+    The IncreasingWidthDiscretiser() first finds the boundaries for the intervals for
+    each variable. Then, it transforms the variables, that is, sorts the values into
+    the intervals.
+
+    More details in the :ref:`User Guide <increasing_width_discretiser>`.
+
+    Parameters
+    ----------
+    {variables}
+
+    bins: int, default=10
+        Desired number of intervals / bins.
+
+    {return_object}
+
+    {return_boundaries}
+
+    Attributes
+    ----------
+    {binner_dict_}
+
+    {variables_}
+
+    {feature_names_in_}
+
+    {n_features_in_}
+
+    Methods
+    -------
+    {fit}
+
+    {fit_transform}
+
+    {transform}
+
+    References
+    ----------
+    .. [1]  J. Reiser, "Classification Systems",
+        https://www.slideshare.net/johnjreiser/classification-systems
```

Comment:
Would it be worth adding additional resources? I haven't heard of this method, so I had to search. I found these which could be useful:
http://wiki.gis.com/wiki/index.php/Geometric_Interval_Classification
https://en.wikipedia.org/wiki/Geometric_progression

Also, it sounds like it is used for GIS data. Is this correct? If yes, maybe you could add a note on this either here or in the user guide?

Based on this, and according to the name the person uses in the slides you shared, would it not be better to call this transformer **GeometricIntervalDiscretiser**? Instead of increasingWidth?

If we use the same nomenclature, then users find it easier to join the dots and find more documentation and examples online.

Reply:
`EqualWidth`
`EqualFrequency`
`IncreasingWidth`

This seems more consistent than:

`EqualWidth`
`EqualFrequency`
`GeometricInterval`

But I don't know

Reply:
GeomtricWidth?